src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -46,100 +65,159 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 		});
 	}
 
-	private async loadBtcTransactionsData({ btcAddress }: { btcAddress: BtcAddress }): Promise<{
-		newTransactions: BitcoinTransaction[];
-		latestBitcoinBlockHeight: number | undefined;
-	}> {
+	private async loadBtcTransactionsData({
+		btcAddress
+	}: {
+		btcAddress: BtcAddress;
+	}): Promise<CertifiedData<BtcTransactionUi>[]> {
 		try {
 			const { txs: fetchedTransactions } = await btcAddressData({ btcAddress });
 
 			const newTransactions = fetchedTransactions.filter(({ hash }) =>
 				isNullish(this.store.transactions[`${hash}`])
 			);
 
-			this.store = {
-				...this.store,
-				transactions: {
-					...this.store.transactions,
-					...newTransactions.reduce(
-						(acc, transaction) => ({
-							...acc,
-							[transaction.hash]: {
-								certified: false,
-								data: transaction
-							}
-						}),
-						{}
-					)
-				}
-			};
-
 			const latestBitcoinBlockHeight = await btcLatestBlockHeight();
 
-			return { newTransactions, latestBitcoinBlockHeight };
+			return nonNullish(latestBitcoinBlockHeight)
```

Comment:
Actually, it's a good point - the `isNullish(latestBitcoinBlockHeight)` is redundant, the value is expected to always be a number. I think that check is left over from the time when we used another API for getting current block height and where it could have been nullish. Updated now.